service/src/main/java/bio/terra/pipelines/stairway/SubmitQuotaConsumedSubmissionStep.java

```diff
@@ -0,0 +1,134 @@
+              null, "quotaConsumed", "quota_consumed", PipelineVariableTypesEnum.INTEGER));
+
+  private final Logger logger = LoggerFactory.getLogger(SubmitQuotaConsumedSubmissionStep.class);
+
+  public SubmitQuotaConsumedSubmissionStep(RawlsService rawlsService, SamService samService) {
+    this.samService = samService;
+    this.rawlsService = rawlsService;
+  }
+
+  @Override
+  @SuppressWarnings(
+      "java:S2259") // suppress warning for possible NPE when calling pipelineName.getValue(),
+  //  since we do validate that pipelineName is not null in `validateRequiredEntries`
+  public StepResult doStep(FlightContext flightContext) {
+    // validate and extract parameters from input map
+    FlightMap inputParameters = flightContext.getInputParameters();
+    FlightUtils.validateRequiredEntries(
+        inputParameters,
+        JobMapKeys.PIPELINE_NAME,
+        ImputationJobMapKeys.CONTROL_WORKSPACE_BILLING_PROJECT,
+        ImputationJobMapKeys.CONTROL_WORKSPACE_NAME,
+        ImputationJobMapKeys.WDL_METHOD_VERSION,
+        ImputationJobMapKeys.PIPELINE_INPUT_DEFINITIONS);
+
+    PipelinesEnum pipelineName = inputParameters.get(JobMapKeys.PIPELINE_NAME, PipelinesEnum.class);
+    String controlWorkspaceName =
+        inputParameters.get(ImputationJobMapKeys.CONTROL_WORKSPACE_NAME, String.class);
+    String controlWorkspaceProject =
+        inputParameters.get(ImputationJobMapKeys.CONTROL_WORKSPACE_BILLING_PROJECT, String.class);
+    String wdlMethodVersion =
+        inputParameters.get(ImputationJobMapKeys.WDL_METHOD_VERSION, String.class);
+    List<PipelineInputDefinition> inputDefinitions =
+        inputParameters.get(
+            ImputationJobMapKeys.PIPELINE_INPUT_DEFINITIONS, new TypeReference<>() {});
+
+    // validate and extract parameters from working map
+    FlightMap workingMap = flightContext.getWorkingMap();
+
+    RawlsSubmissionStepHelper rawlsSubmissionStepHelper =
+        new RawlsSubmissionStepHelper(
+            rawlsService, samService, controlWorkspaceProject, controlWorkspaceName, logger);
+
+    Optional<StepResult> validationResponse =
+        rawlsSubmissionStepHelper.validateRawlsSubmissionMethodHelper(
+            QUOTA_CONSUMED_METHOD_NAME,
+            wdlMethodVersion,
+            inputDefinitions,
+            QUOTA_CONSUMED_OUTPUT_DEFINITION_LIST,
+            pipelineName);
+
+    // if there is a validation response that means the validation failed so return it
+    if (validationResponse.isPresent()) {
+      return validationResponse.get();
+    }
+
+    // create submission request
+    SubmissionRequest submissionRequest =
+        new SubmissionRequest()
+            .entityName(flightContext.getFlightId())
+            .entityType(
+                pipelineName.getValue()) // this must match the configuration the method is set to
+            // launch with.  Will be addressed in
+            // https://broadworkbench.atlassian.net/browse/TSPS-301
```

Comment:
TSPS-301 is closed duplicate of TSPS-303, which is complete. can we remove this comment or is there still work to be done that wasn't done with TSPS-303? https://broadworkbench.atlassian.net/browse/TSPS-303

(I'm not really sure what the issue is here that needs to be addressed)

Reply:
oh yea this is solved - will remove.   This is part of the "magic" validation/updating of the method configuration